### PR TITLE
fix: download image no longer opens a new tab

### PR DESCRIPTION
**Pull request by Cyberxon**

### CODE REVIEW

File: packages/rich-text/src/plugins/shared/FetchingWrappedAssetCard.tsx [hunk 1]
```diff
@@ -30,8 +30,24 @@ interface FetchingWrappedAssetCardProps {
...
+    method: 'GET',
+    headers: {},
+  })
+    .then((response) => {
+      response.arrayBuffer().then(function (buffer) {
+        const url = window.URL.createObjectURL(new Blob([buffer]));
+        const link = document.createElement('a');
+        link.href = url;
+        link.setAttribute('download', fileName); //or any other extension
+        document.body.appendChild(link);
```
Comment (z0al) [inline packages/rich-text/src/plugins/shared/FetchingWrappedAssetCard.tsx:44]: don't we want to remove this new `a` element afterward?

File: packages/rich-text/src/plugins/shared/FetchingWrappedAssetCard.tsx [hunk 2]
```diff
@@ -30,8 +30,24 @@ interface FetchingWrappedAssetCardProps {
   sdk: FieldExtensionSDK;
 }
 
-function downloadAsset(url: string) {
-  window.open(url, '_blank', 'noopener,noreferrer');
+function downloadAsset(url: string, fileName: string) {
+  fetch(url, {
```
Comment (z0al) [inline packages/rich-text/src/plugins/shared/FetchingWrappedAssetCard.tsx:34]: I think this won't work when the target has [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) disabled, right?

File: packages/rich-text/src/plugins/shared/FetchingWrappedAssetCard.tsx [hunk 3]
```diff
@@ -30,8 +30,24 @@ interface FetchingWrappedAssetCardProps {
...
+        const url = window.URL.createObjectURL(new Blob([buffer]));
+        const link = document.createElement('a');
+        link.href = url;
+        link.setAttribute('download', fileName); //or any other extension
+        document.body.appendChild(link);
+        link.click();
+      });
+    })
+    .catch((err) => {
+      console.log(err);
```
Comment (djagya) [inline packages/rich-text/src/plugins/shared/FetchingWrappedAssetCard.tsx:49]: Maybe show a Froam36 notification like "Failed to download the asset" AND re-throw the error so the app can catch it and report to Sentry. Worth checking if throwing an error crashes only the asset card or the whole app.

File: packages/rich-text/src/plugins/shared/FetchingWrappedAssetCard.tsx [hunk 4]
```diff
@@ -30,8 +30,27 @@ interface FetchingWrappedAssetCardProps {
...
+        const link = document.createElement('a');
+        link.href = url;
+        link.setAttribute('download', fileName); //or any other extension
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+      });
+    })
+    .catch((err) => {
+      Notification.error('Failed to dowmload asset');
```
Comment (z0al) [inline packages/rich-text/src/plugins/shared/FetchingWrappedAssetCard.tsx:51]:
```suggestion
      Notification.error('Failed to download asset');
```

**Review by z0al — APPROVED** [review]

**Pull request merged**